### PR TITLE
cctz: enable tools by default + fix iOS/tvOS/watchOS build + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
uhuh, test package was broken with build_tools enabled

File: recipes/cctz/all/conanfile.py [hunk 1]
```diff
@@ -20,53 +21,65 @@ class CCTZConan(ConanFile):
...
-        if self.settings.os == "Windows" and \
-           self.settings.compiler == "Visual Studio" and \
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio" and \
            tools.Version(self.settings.compiler.version) < 14:
-           raise ConanInvalidConfiguration("CCTZ requires MSVC >= 14")
+            raise ConanInvalidConfiguration("CCTZ requires MSVC >= 14")
```
Comment (prince-chrismc) [inline recipes/cctz/all/conanfile.py:51]:
```suggestion
               raise ConanInvalidConfiguration("CCTZ requires MSVC >= 14")
```

Comment (SpaceIm) [reply recipes/cctz/all/conanfile.py:51]: ?

Comment (prince-chrismc) [reply recipes/cctz/all/conanfile.py:51]: The whitespace indent is off... right?

Comment (SpaceIm) [reply recipes/cctz/all/conanfile.py:51]: I've fixed indentation, your suggestion breaks it

Comment (prince-chrismc) [reply recipes/cctz/all/conanfile.py:51]: Ahhhhhh thank you for your patients... looking the the while file I seen the line break in the if above so yes... sorry the one spacebar looked out of place in the diff

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by SpaceIm** [comment]
PRs submitted upstream for iOS and dll installation:
https://github.com/google/cctz/pull/193
https://github.com/google/cctz/pull/194

**Review by SSE4 — APPROVED** [review]

**Pull request merged**